spacy/language.py

```diff
@@ -1675,6 +1677,7 @@ def _link_components(self) -> None:
         # here :(
         for i, (name1, proc1) in enumerate(self.pipeline):
             if isinstance(proc1, ty.ListenedToComponent):
+                proc1.listener_map = {}
```

Comment:
Perhaps we could put this functionality behind a `reset` flag that is `False` by default? It's really only relevant when deleting pipes, right? Otherwise we can just have the incremental adding behaviour as before?

I don't care strongly, it probably won't have much impact either way as we're looping through everything anyway, and only preventing `append` when the listener had already been recorded, so the traversal will be equally long either way. It's just that we're kind of reconstructing the same map every time again which feels wasteful.

Reply:
I think it's such a small number of things and it's only called when pipes are added or removed, so there's no real benefit to making this marginally faster?